### PR TITLE
change put to patch for user change password route and add useGuard to all user route

### PR DESCRIPTION
**Pull request by Muh-Baharuddin**
@sidratul please review my code for user route in controller

### CODE REVIEW

File: src/users/users.controller.ts [hunk 1]
```diff
@@ -43,6 +45,7 @@ export class UsersController {
   }
 
   @Delete(':id')
+  @UsePipes(ValidationPipe)
```
Comment (sidratul) [inline src/users/users.controller.ts]: delete user also check JTW token

File: src/users/users.controller.ts [hunk 2]
```diff
@@ -22,16 +22,18 @@ export class UsersController {
   constructor(private readonly usersService: UsersService) {}
 
   @Get()
+  @UseGuards(JwtAuthGuard)
```
Comment (sidratul) [inline src/users/users.controller.ts]: if all method in class should check JWT, put JWT guard in controller

File: src/users/users.controller.ts (hunk 1 above)
Comment (Muh-Baharuddin) [reply src/users/users.controller.ts]: done

File: src/users/users.controller.ts (hunk 2 above)
Comment (Muh-Baharuddin) [reply src/users/users.controller.ts]: done

**Review by sidratul — APPROVED** [review]

**Pull request merged**